cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationVerificationStage.java

```diff
@@ -144,8 +147,18 @@ protected void verifyData(ApplicationId appId,
         throw new RuntimeException(result.getMessage());
       }
       String dsName = dataSetCreateSpec.getInstanceName();
-      DatasetId datasetInstanceId = appId.getParent().dataset(dsName);
-      DatasetSpecification existingSpec = dsFramework.getDatasetSpec(datasetInstanceId);
+      final DatasetId datasetInstanceId = appId.getParent().dataset(dsName);
+      // get the authorizing user
+      String authorizingUser =
+        AuthorizationUtil.getAppAuthorizingUser(ownerAdmin, authenticationContext, appId,
+                                                ownerPrincipal == null ? null : ownerPrincipal.getPrincipal());
```

Comment:
We can just pass the `KerberosPrincipalId` to the method and do the check there, so that we don't need this check in multiple places.